### PR TITLE
TRUNK-4839: Exposing Patient Search Options Via UI

### PR DESCRIPTION
**Pull request by Nishen-Peiris**
## Description
01.Remove unused import org.hibernate.HibernateException
02.Remove 1 redundant type arguments in the new expression at org.openmrs.api.db.hibernate.HibernatePatientDAO#sortDuplicatePatients
03.Remove 2 redundant type arguments in new expressions at org.openmrs.api.db.hibernate.HibernatePatientDAO#getPatients(String, boolean, Integer, Integer)
04.Remove 7 redundant type arguments in the new expression at org.openmrs.api.db.hibernate.HibernatePatientDAO#getDuplicatePatientsSQLString

## Related Issue
I haven't implemented the feature completely.
see https://issues.openmrs.org/browse/TRUNK-4839

## Checklist:
- [x] My pull request only contains one single commit.
- [x] My pull request is based on the latest master branch
  `git pull --rebase upstream master`.
- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.
- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java [hunk 1]
```diff
@@ -192,7 +191,7 @@ private void insertPatientStubIfNeeded(Patient patient) {
 		criteriaExactMatch = new PatientSearchCriteria(sessionFactory, criteriaExactMatch).prepareCriteria(query, true,
 		    true, includeVoided);
 		
-		Set<Patient> patients = new LinkedHashSet<Patient>();
+		Set<Patient> patients = new LinkedHashSet<>();
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: Do you mind sharing the reason behind removing Patient?

Comment (Nishen-Peiris) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: @dkayiwa, Its a new feature introduced by Java 7 know as the "Diamond" operator. For more info, follow http://openjdk.java.net/projects/jdk7/features/#f618

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernatePatientDAO.java]: @Nishen-Peiris oh i see! :)
The ticket under which you are creating these pull requests does not seem related to the changes you are making. Do you mind creating a new ticket? :)

**Comment by dkayiwa** [comment]
@Nishen-Peiris thanks again for all that you are doing! 👍 
Since you already did this, i have gone ahead and merged it. You will create a new ticket about the code improvements that you are doing and then the upcoming pull requests will be based on it. :)

**Comment by Nishen-Peiris** [comment]
@dkayiwa thanks...
